waku/node/peer_manager/peer_manager.nim

```diff
@@ -87,6 +91,7 @@ type PeerManager* = ref object of RootObj
   colocationLimit*: int
   started: bool
   shardedPeerManagement: bool # temp feature flag
+  onConnectionChange*: ConnectionChangeHandler
```

Comment:
I think it is better to enforce this parameter to the `new` proc and then we can make sure it is always `not isNil():`
```suggestion
  onConnectionChange: ConnectionChangeHandler
```